app/components/search/search_bar_component.rb

```diff
@@ -4,34 +4,69 @@ class Search::SearchBarComponent < ViewComponent::Base
   def initialize(
     resource: nil,
     query: nil,
-    search_path: "/search",
-    resource_options: %w[Users Projects],
-    options: {}
+    sort_by: nil,
+    sort_direction: nil
   )
     super
     @resource = resource
     @query = query
-    @search_path = search_path
-    @resource_options = resource_options
-    @placeholders = options[:placeholders] || default_placeholders
-    @option_labels = options[:option_labels] || default_option_labels
+    @sort_by = sort_by || default_sort_by
+    @sort_direction = sort_direction || default_sort_direction
+  end
+
+  def all_sorting_options
+    {
+      "Users" => sorting_options_for_users,
+      "Projects" => sorting_options_for_projects
+    }
   end
 
   private
 
-    attr_reader :resource, :query, :search_path, :resource_options, :placeholders, :option_labels
+    attr_reader :resource, :query, :sort_by, :sort_direction
+
+    def resource_options
+      %w[Users Projects]
+    end
 
-    def default_placeholders
+    def placeholders
       {
         "Users" => I18n.t("components.search_bar.placeholders.users"),
         "Projects" => I18n.t("components.search_bar.placeholders.projects")
       }
     end
 
-    def default_option_labels
+    def option_labels
       {
         "Users" => I18n.t("components.search_bar.options.users"),
         "Projects" => I18n.t("components.search_bar.options.projects")
       }
     end
+
+    def sorting_options_for_users
+      [
+        { value: "created_at", label: I18n.t("components.search_bar.sorting.users.join_date") },
+        { value: "total_circuits", label: I18n.t("components.search_bar.sorting.users.total_circuits") }
+      ]
+    end
+
+    def sorting_options_for_projects
+      [
+        { value: "created_at", label: I18n.t("components.search_bar.sorting.projects.created_date") },
+        { value: "views", label: I18n.t("components.search_bar.sorting.projects.views") },
+        { value: "stars", label: I18n.t("components.search_bar.sorting.projects.stars") }
+      ]
```

Comment:
Suggestion(non-blocking): use .freeze and also define the constant on top of the file